islandora.api.php

```diff
@@ -886,3 +886,41 @@ function hook_islandora_edit_datastream_registry_alter(&$edit_registry, $context
 function hook_islandora_repository_connection_construction_alter(RepositoryConnection $instance) {
   $instance->userAgent = "Tuque/cURL";
 }
+
+/**
+ * Allow a overridable backend for generating breadcrumbs.
+ *
+ * Stolen shamelessly from @adam-vessey.
+ *
+ * @return array
+ *   Should return an associative array mapping unique (module-prefixed,
+ *   preferably) keys to associative arrays containing:
+ *   - title: A human-readable title for the backend.
+ *   - callable: A PHP callable to call for this backend, implementing
+ *     callback_islandora_basic_collection_query_backends().
+ *   - file: An optional file to load before attempting to call the callable.
+ */
+function hook_islandora_breadcrumbs_backends() {
+  return array(
+    'awesome_backend' => array(
+      'title' => t('Awesome Backend'),
+      'callable' => callback_islandora_breadcrumbs_backends($object),
+    ),
+  );
+}
+
+/**
+ * Generate an array of links for breadcrumbs leading to $object, root first.
+ *
+ * Stolen shamelessly from @adam-vessey.
+ *
+ * @param AbstractObject $object
+ *   The object to generate breadcrumbs for.
+ *
+ * @return array
+ *   Array of links from root to $object.
+ */
+function callback_islandora_breadcrumbs_backends(AbstractObject $object) {
+  // Do something to get an array of breadcrumb links for $object, root first.
+  return array($root_link, $collection_link, $object_link);
```

Comment:
Something of an ambiguity, in both the example and documentation: Are we to include the link to the object itself? It's done in the recursive implementation ([but popped off before rendering](https://github.com/Islandora/islandora/pull/651/files#diff-79500f6fd0f804357e071e8c33c48560R34))... I'm thinking we should probably exclude it?